AesCbcWithIntegrity.java

```diff
@@ -580,6 +581,21 @@ public boolean equals(Object obj) {
     }
 
     /**
+     * Copy the elements from the start to the end
+     *
+     * @param from  the source
+     * @param start the start index to copy
+     * @param end   the end index to finish
+     * @return the new buffer
+     */
+    private static byte[] copyOfRange(byte[] from, int start, int end) {
+        int length = end - start;
```

Comment:
This will not behave correctly if `end - start` is too large.

You shouldn't copy from here, but the Android implementation can be found at https://android.googlesource.com/platform/libcore/+/android-5.0.0_r1/luni/src/main/java/java/util/Arrays.java